docs/CImportNameTranslation.md

```diff
@@ -0,0 +1,191 @@
+
+# Enums
+
+1. Anonymous? Import elements as constants of the underlying type, *except* that Int is used for an inferred underlying type if all the cases fit in an Int32, since integer conversions are explicit in Swift.
+2. `ns_error_domain` attribute? Import as an error struct (see below)
+3. `flag_enum` attribute? Import as an option set struct (see below)
+4. `enum_extensibility` attribute? Import as an `@objc` enum, whether open or closed
+5. Otherwise, import as a RawRepresentable struct _without renaming constants_ (this is probably an oversight but would be source-breaking to change)
+
+An enum declared as `typedef enum { … } MyEnum;` is not considered "anonymous"; the typedef name is used instead. This interpretation has precedent in the C++ linkage rules, which have a similar special case for typedefs of unnamed enums (C++17 [dcl.typedef]p9).
+
+## `@objc` enums
+
+```objc
+enum TimeOfDay __attribute__((enum_extensibility(open))) : long {
+  TimeOfDayMorning,
+  TimeOfDayAfternoon,
+  TimeOfDayNight,
+  TimeOfDayEvening = TimeOfDayNight
+};
+// Most commonly seen as NS_ENUM and NS_CLOSED_ENUM.
+```
+
+```swift
+@objc enum TimeOfDay: Int {
+  init?(rawValue: Int)
+  var rawValue: Int { get }
+
+  case morning
+  case afternoon
+  case night
+
+  static var evening: TimeOfDay { get }
+}
+```
+
+- Enum name is enum name.
+- Underlying type becomes raw type.
+- Case names follow the rules for "enum-style prefix stripping" below.
+- `init?(rawValue:)` currently does not check cases, but probably should for "closed" enums (Clang's equivalent of `@frozen`)
+
+In order to deal with cases with the same underlying value, the importer picks a set of _canonical cases_ by walking the enum and recording the first *available* enumerator with a particular value. (Note that deprecated cases are still considered available unless they were deprecated for a platform older than Swift's earliest deployment targets, OS X 10.9 and iOS 7.) Any "non-canonical" cases are imported as computed properties instead.
+
+## Error enum structs
+
+```objc
+enum VagueFailureCode __attribute__((ns_error_domain(VagueFailureDomain))) : long {
+  VagueFailureBadness,
+  VagueFailureWorseness,
+  VagueFailureWorstness
+};
+// Most commonly seen as NS_ERROR_ENUM.
+```
+
+```swift
+struct VagueFailure: Error {
+  @objc enum Code: Int {
+    init?(rawValue: Int)
+    var rawValue: Int { get }
+
+    case badness
+    case worseness
+    case worstness
+
+    typealias ErrorType = VagueFailure
+  }
+
+  static var badness: VagueFailure.Code { get }
+  static var worseness: VagueFailure.Code { get }
+  static var worstness: VagueFailure.Code { get }
+
+  static var errorDomain: String { get }
+}
+```
+
+- Struct name is enum name, optionally dropping the word "Code".
```

Comment:
Is this optional to Omit-needless-words?

Reply:
Whoops. I should probably just say "dropping the suffix `"Code"` if present".